Source/CBLMultipartWriter.m

```diff
@@ -109,10 +109,14 @@ - (CBLStatus) addAttachment: (CBL_Attachment*)attachment {
                                                              andLength: &contentLength];
     if (!contentStream)
         return kCBLStatusAttachmentNotFound;
+    
     uint64_t declaredLength = attachment.possiblyEncodedLength;
-    if (declaredLength != 0 && contentLength != declaredLength)
+    if (attachment.isEncrypted)
```

Comment:
I think it's better to test for `contentLength==0` instead of encryption. `-getContentStreamDecoded:andLength:` returns a zero length to mean it doesn't know the length; for now that happens because of encryption, but it's possible that there could be other reasons.